src/main/java/com/icthh/xm/uaa/service/UserService.java

```diff
@@ -170,6 +170,49 @@ public User createUser(UserDTO user) {
             .map(UserDTO::new);
     }
 
+    /**
+     * Blocks user account. Throws BusinessException.ERROR_USER_BLOCK_HIMSELF for self-block
+     * @param userKey - user key
+     * @return - user data
+     */
+    public Optional<UserDTO> blockUserAccount(String userKey) {
+        if (xmAuthenticationContextHolder.getContext().getRequiredUserKey().equals(userKey)) {
+            throw new BusinessException(ERROR_USER_BLOCK_HIMSELF, "Forbidden to block himself");
+        }
+        return changeUserAccountState(userKey, Boolean.FALSE);
+    }
+
+    /**
+     * Activates user account. Throws BusinessException.ERROR_USER_BLOCK_HIMSELF for self-block
+     * @param userKey - user key
+     * @return - user data
+     */
+    public Optional<UserDTO> activateUserAccount(String userKey) {
```

Comment:
add lep annotation please.
for send email for example

Reply:
@LogicExtensionPoint("ActivateUserAccount")